### PR TITLE
Problem: Adding chain-maind keys will not be actually stored in disk …

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add decoding _hexcode in service name fetched

### CODE REVIEW

**Comment by leejw51crypto** [comment]
keyring pr is here
https://github.com/99designs/keyring/pull/83/files

**Comment by JayT106** [comment]
I see some of the modules been removed from the `go.sum`. How do they relate to the fixes?
For the keyring dependency, is it better to merge @leejw51crypto 's fixes to the upstream repo first and then use it? (Not sure how urgent is the fixes)

**Comment by tomtau** [comment]
yeah, may be better to wait for the upstream -- it seems it's maintained

**Comment by leejw51crypto** [comment]
> I see some of the modules been removed from the `go.sum`. How do they relate to the fixes?
> For the keyring dependency, is it better to merge @leejw51crypto 's fixes to the upstream repo first and then use it? (Not sure how urgent is the fixes)

i ran "go mod tidy" to check mod

this is urgent issue, because in linux,  keyring is not usable by default.

**Review by tomtau — CHANGES_REQUESTED** [review]
there are some conflicts with the latest master.

it seems upstream may take time -- I've forked the repo with the fix here, so the go.mod can be updated to this one: https://github.com/crypto-org-chain/keyring/tree/gnome for the moment

**Comment by leejw51crypto** [comment]
ok, i'll rebase with the latest

File: go.mod [hunk 1]
```diff
@@ -29,3 +29,5 @@ replace google.golang.org/grpc => google.golang.org/grpc v1.33.2
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
 replace github.com/cosmos/ledger-cosmos-go => github.com/crypto-com/ledger-cosmos-go v0.9.10-0.20200929055312-01e1d341de0f
+
+replace github.com/99designs/keyring => github.com/leejw51crypto/keyring v1.1.6-gnome
```
Comment (tomtau) [inline go.mod:33]:
```suggestion
replace github.com/99designs/keyring => github.com/crypto-org-chain/keyring v1.1.6-gnome
```

**Comment by leejw51crypto** [comment]
inspecting integration test failures

**Comment by leejw51crypto** [comment]
fixed, increase waiting time twice :-)

**Review by tomtau — APPROVED** [review]

**Review by matias904 — CHANGES_REQUESTED** [review]
1150696e9218c9c6b615b3fdd0de1d1ccc8c1fbb

**Pull request merged**